database/factories/UserFactory.php

```diff
@@ -1,34 +1,41 @@
 <?php
 
-/** @var \Illuminate\Database\Eloquent\Factory $factory */
+namespace Database\Factories;
 
+use App\Contracts\AccountRepositoryInterface;
 use App\User;
-use Faker\Generator as Faker;
+use Illuminate\Database\Eloquent\Factories\Factory;
 use Illuminate\Support\Str;
 
-/*
-|--------------------------------------------------------------------------
-| Model Factories
-|--------------------------------------------------------------------------
-|
-| This directory should contain each of the model factory definitions for
-| your application. Factories provide a convenient way to generate new
-| model instances for testing / seeding your application's database.
-|
-*/
+class UserFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = User::class;
 
-$factory->define(User::class, function (Faker $faker) {
-    return [
-        'name' => $faker->name,
-        'email' => $faker->unique()->safeEmail,
-        'email_verified_at' => now(),
-        'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password
-        'remember_token' => Str::random(10),
-    ];
-});
+    public function configure()
+    {
+        return $this->afterCreating(function (User $user) {
+            $user->account()->save(static::factoryForModel(resolve(AccountRepositoryInterface::class)->class())->make());
```

Comment:
I'm curious why `static::` here instead of `Factory::`, as we have everywhere else. In this particular case, both would resolve to the same method in `Illuminate\Database\Eloquent\Factories\Factory`, correct?

Reply:
That's correct, since we are already in a Factory. `self::` would also work. But maybe it is better to use `Factory` to be clear what the intent is.